### PR TITLE
TMEDIA-150 - Add bandito support to promos

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Add the bandito registerSuccessEvent to PromoHeadline and PromoImage component, and update all components to utilize the shared Headline and Image components

## Jira Ticket
- [TMEDIA-150](https://arcpublishing.atlassian.net/browse/TMEDIA-150)

## Acceptance Criteria
1. Customers can run Bandito tests on the following Arc Blocks: 
    * XL Promo Block
    * L Promo Block
    * Medium Promo Block
    * Small Promo Block
    * XL Manual Promo Block
    * L Manual Promo Block
    * Medium Manual Promo Block
    * Small Manual Promo Block

## Test Steps

1. Checkout this branch `git checkout TMEDIA-150-add-banito-to-promos`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/medium-promo-block,@wpmedia/small-promo-block,@wpmedia/extra-large-manual-promo-block,@wpmedia/large-manual-promo-block,@wpmedia/medium-manual-promo-block,@wpmedia/small-manual-promo-block,@wpmedia/shared-styles`
3.  Verify promo blocks didn't change - test page - http://localhost/pf/test-all-promos/?_website=the-gazette

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working.
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-import getProperties from 'fusion:properties';
 import { useFusionContext } from 'fusion:context';
-import { imageRatioCustomField, ratiosFor } from '@wpmedia/resizer-image-block';
+import { useEditableContent } from 'fusion:content';
+import { LazyLoad, isServerSide } from '@wpmedia/engine-theme-sdk';
+import { imageRatioCustomField } from '@wpmedia/resizer-image-block';
+import { PromoHeadline, PromoImage } from '@wpmedia/shared-styles';
 
 import '@wpmedia/shared-styles/scss/_extra-large-promo.scss';
-import { Image, LazyLoad, isServerSide } from '@wpmedia/engine-theme-sdk';
```
Comment (JackHowa) [inline blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:10]: thanks for fixing the ordering

File: blocks/shared-styles/scss/_medium-promo.scss [hunk 2]
```diff
@@ -4,6 +4,7 @@
   a {
     color: $ui-primary-font-color;
     position: relative;
+    text-decoration: none;
   }
 
   a > picture > img {
```
[the remark this answers is not in the document]
Comment (matthewroach) [reply blocks/shared-styles/scss/_medium-promo.scss]: Fix coming

File: blocks/shared-styles/_children/promo-image/index.jsx [hunk 3]
```diff
@@ -65,6 +65,7 @@ const PromoImage = ({
       rel={newTab ? 'noreferrer noopener' : ''}
       aria-hidden="true"
       tabIndex="-1"
+      onClick={registerSuccessEvent}
```
Comment (JackHowa) [inline blocks/shared-styles/_children/promo-image/index.jsx:68]: looks clear enough. did they give us any docs or something?

File: blocks/shared-styles/_children/promo-image/index.jsx [hunk 4]
```diff
@@ -65,6 +65,7 @@ const PromoImage = ({
       rel={newTab ? 'noreferrer noopener' : ''}
       aria-hidden="true"
       tabIndex="-1"
```
Comment (JackHowa) [inline blocks/shared-styles/_children/promo-image/index.jsx:67]: ew, do we need this -1?

File: blocks/shared-styles/scss/_small-promo.scss [hunk 5]
```diff
@@ -1,7 +1,13 @@
 .small-promo {
-
   color: $ui-primary-font-color;
   margin-top: 0;
+
+  @media screen and (min-width: map-get($grid-breakpoints, 'sm')) {
+    .row {
+      grid-gap: 1rem;
```
Comment (JackHowa) [inline blocks/shared-styles/scss/_small-promo.scss:7]: nice nice. one step closer to our grid promo universal component

**Review by JackHowa — APPROVED** [review]
Works as expected with passing in the data

File: blocks/shared-styles/_children/promo-image/index.jsx (hunk 3 above)
Comment (matthewroach) [reply blocks/shared-styles/_children/promo-image/index.jsx:68]: I used this - https://corecomponents.arcpublishing.com/alc/arc-products/pagebuilder/user-docs/pagebuilder-editor-content-testing-guide/

**Pull request merged**